packages/universal-components/src/Button/ButtonInner.js

```diff
@@ -78,24 +81,38 @@ export default function ButtonInner({
         circled && styleSheet.buttonCircled,
         sizeTheme(size, leftIcon != null, rightIcon != null).buttonSizeWrapper,
         style,
+        isLoading && styleSheet.loadingState,
       ]}
       testID={testID}
     >
-      <View style={layout.row}>
-        {leftIcon != null && <View style={rightSpace}>{leftIcon}</View>}
-        {children ||
-          (label != null && (
-            <ButtonTitle text={label} type={type} size={size} />
-          ))}
-      </View>
-      <View style={layout.row}>
-        {sublabel != null && (
+      {isLoading ? (
```

Comment:
What about passing this props to loader and from Loader to `ActivityIndicator` it has the animating props, which will hide the loader if false. [ActivityIndicator](https://facebook.github.io/react-native/docs/activityindicator#animating)

Reply:
Just pushed the changes. Did you mean this?

Reply:
Yes, looks better, now we got rid of the `isLoading ? ... : ...` from ButtonInner, so it is much more readable 🎉